Gems/ROS2/Code/Source/RobotImporter/SDFormat/Hooks/ROS2SkidSteeringModelHook.cpp

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <ROS2/Frame/ROS2FrameComponent.h>
+#include <RobotControl/Controllers/SkidSteeringController/SkidSteeringControlComponent.h>
+#include <RobotControl/ROS2RobotControlComponent.h>
+#include <RobotImporter/SDFormat/ROS2ModelPluginHooks.h>
+#include <RobotImporter/SDFormat/ROS2SDFormatHooksUtils.h>
+#include <Source/EditorArticulationLinkComponent.h>
+#include <Source/EditorHingeJointComponent.h>
+#include <VehicleDynamics/ModelComponents/SkidSteeringModelComponent.h>
+#include <VehicleDynamics/Utilities.h>
+
+#include <sdf/Joint.hh>
+#include <sdf/Link.hh>
+#include <sdf/Model.hh>
+#include <sdf/Plugin.hh>
+
+namespace ROS2::SDFormat
+{
+    namespace Parser
+    {
+        AZ::EntityId GetAxleWheelId(const std::string& jointName, const sdf::Model& sdfModel, const CreatedEntitiesMap& createdEntities)
+        {
+            const auto jointPtr = sdfModel.JointByName(jointName);
+            if (jointPtr != nullptr)
+            {
+                const auto linkName(jointPtr->ChildName().c_str());
+                const auto linkPtr = sdfModel.LinkByName(linkName);
+                if (linkPtr != nullptr && createdEntities.contains(linkPtr))
+                {
+                    const auto& entityResult = createdEntities.at(linkPtr);
```

Comment:
Is it possible for the entityResult to not be success here?

Reply:
It should not happen, but we had segmentation faults in similar cases due to name collisions lately (name collisions leading to not creating all entities). It is always safer to add a check.